### PR TITLE
Add predicates

### PR DESCRIPTION
**Pull request by fkrause98**
## Changes:
- Add state flags to the vm state.
- Conditionally execute opcodes depending on instructions.
- Add tests for features.

### CODE REVIEW

File: src/lib.rs
Comment (gianbelinche) [inline src/lib.rs:1]: Why did you add VMState as a return value? Its never used anywhere

**Review by gianbelinche — CHANGES_REQUESTED** [review]
Just that one comment, everything else is fine

File: src/lib.rs [hunk 1]
```diff
@@ -14,14 +14,20 @@ use zkevm_opcode_defs::ISAVersion;
 use zkevm_opcode_defs::LogOpcode;
 use zkevm_opcode_defs::Opcode as Variant;
 
-pub fn run_program(bin_path: &str) -> U256 {
+pub fn run_program(bin_path: &str) -> (U256, VMState) {
```
Comment (gianbelinche) [inline src/lib.rs:17]: Why did you add VMState as a return value? Its never used anywhere

Comment (fkrause98) [reply src/lib.rs:17]: I though it would be useful to inspect the VM's final state when testing.
Let's say we want to test if an instruction modifies the "eq" flag, I can't come up with another way to test it.

**Review by gianbelinche — APPROVED** [review]

**Review by juan518munoz — APPROVED** [review]
LGTM

File: src/lib.rs (hunk 1 above)
Comment (juan518munoz) [reply src/lib.rs:17]: Agree that being able to inspect the `VMState` is useful for testing, it comes in handy for the (yet to be pushed) PR for [arithmetic instructions](https://github.com/matter-labs/zksync-era/blob/main/docs/specs/zk_evm/vm_specification/zkSync_era_virtual_machine_primer.md#arithmetic-instructions).
